src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -42,28 +56,168 @@ public void setFrameworkId(FrameworkID frameworkId) throws InterruptedException,
     zkState.store(value).get();
   }
 
-  public Node getNamenode() throws ClassNotFoundException, InterruptedException,
-      ExecutionException, IOException {
-    return get(NAMENODE_KEY);
+  public List<String> getDeadJournalNodes() {
+    HashMap<String, String> journalNodes = getJournalNodes();
+    Set<String> journalHosts = journalNodes.keySet();
+    List<String> deadJournalHosts = new ArrayList<>();
+
+    for (String journalHost: journalHosts) {
+      if (journalNodes.get(journalHost) == null) {
+        deadJournalHosts.add(journalHost);
+      }
+    }
+    return deadJournalHosts;
+  }
+  public List<String> getDeadNameNodes() {
+    HashMap<String, String> nameNodes = getNameNodes();
+    Set<String> nameHosts = nameNodes.keySet();
+    List<String> deadNameHosts = new ArrayList<>();
+
+    for (String nameHost : nameHosts) {
+      if (nameNodes.get(nameHost) == null) {
+        deadNameHosts.add(nameHost);
+      }
+    }
+    return deadNameHosts;
   }
+  public List<String> getDeadDataNodes() {
+    HashMap<String, String> dataNodes = getDataNodes();
+    Set<String> dataHosts = dataNodes.keySet();
+    List<String> deadDataHosts = new ArrayList<>();
 
-  public void setNamenode(Node node) throws InterruptedException, ExecutionException, IOException {
-    set(NAMENODE_KEY, node);
+    for (String dataHost : dataHosts) {
+      if (dataNodes.get(dataHost) == null) {
+        deadDataHosts.add(dataHost);
+      }
+    }
+    return deadDataHosts;
+  }
+  // TODO (nicgrayson) add tests with in memory zookeeper
```

Comment:
s/in memory zookeeper/in-memory state implementation/ (since it's just an in-memory hashmap behind the state abstraction, not a real zk impl)